src/Redistribution.sol

```diff
@@ -774,45 +731,81 @@ contract Redistribution is AccessControl, Pausable {
         require(cr == currentRevealRound, "round received no reveals");
         require(cr > currentClaimRound, "round already received successful claim");
 
-        uint256 currentWinnerSelectionSum;
-        bytes32 winnerIs;
-        bytes32 randomNumber;
-        uint256 randomNumberTrunc;
         bytes32 truthRevealedHash;
         uint8 truthRevealedDepth;
-        uint256 revIndex;
-        string memory winnerSelectionAnchor = currentWinnerSelectionAnchor();
-        uint256 k = 0;
+        Reveal memory _winner;
 
         // Get current truth
         (truthRevealedHash, truthRevealedDepth) = getCurrentTruth();
-        uint256 commitsArrayLength = currentCommits.length;
 
-        for (uint256 i = 0; i < commitsArrayLength; i++) {
+        // Evaluate revealers, get Winners and Losers
+        (_winner, , , ) = evaluateRevealers(truthRevealedHash, truthRevealedDepth);
```

Comment:
yes, it will, we could introduce some flag to skip those parts but then again it's a Read function so it doesn't cost any gas and basically nobody would have problem with that. 

This was the basic idea to use one function for both isWinner which is read only and winnerSelection which is state changing and not double the code for both to do the same thing besides slashing/freezing.